OpenRA.Game/FileSystem/MixFile.cs

```diff
@@ -122,6 +122,15 @@ static MemoryStream Decrypt(uint[] h, Blowfish fish)
 
 		static uint[] ReadBlocks(Stream s, long offset, int count)
 		{
+			if (offset < 0)
+				throw new ArgumentOutOfRangeException("offset", "Non-negative number required.");
+
+			if (count < 0)
```

Comment:
That won't be a problem. http://stackoverflow.com/questions/3102353/what-happens-if-i-initialize-an-array-to-size-0